SG14/inplace_function.h

```diff
@@ -84,13 +84,27 @@ template<typename R, typename... Args> struct vtable
       ); }
     },
 		copy_ptr{ [](storage_ptr_t dst_ptr, storage_ptr_t src_ptr) -> void
-		  { new (dst_ptr) C{ (*static_cast<C*>(src_ptr)) }; }
+		  {
+        if constexpr (std::is_trivially_copy_constructible<C>::value)
+          std::memcpy(dst_ptr, src_ptr, sizeof(C));
+        else
+          new (dst_ptr) C{ (*static_cast<C*>(src_ptr)) };
+      }
```

Comment:
Does this buy you anything in performance? I would actually expect the compiler to be *better* at inlining a trivial constructor than at inlining a `memcpy`. That is, I would advise removing lines 88,89,90, and 92, unless you can show that they're doing something useful.
Ditto for the move-constructor and destructor cases below: if the `if constexpr` isn't doing something useful (and I will be surprised if it is), I'd recommend to remove it.

Reply:
I agree.